### ISSUE TITLE
Not respecting version pins

### ISSUE THREAD

**Issue by mcg1969** [Anaconda-Platform/anaconda-project#201]
I have an environment that contains `r-shiny`, `r-dplyr`, `r-curl`, and `r-base=3.5.0`, but not `r-shinydashboard` and `r-leaflet`. This is the `anaconda-project.yml` file:
```
name: 086-bus-dashboard
commands:
  default:
    unix: R -e 'shiny::runApp(appDir = ".", port = 8086, host="0.0.0.0")'
env_specs:
    anaconda50_r:
        packages:
            - r-shiny
            - r-shinydashboard
            - r-leaflet
            - r-dplyr
            - r-curl
            - r-base=3.5.0
```
This is Linux. When I run `anaconda-project prepare`, I am upgraded to `r-base=3.6.0`, despite the pinning. I can blame `anaconda-project` and not `conda`, because the conda command generated is:
```
conda install --yes -n anaconda50_r r-shinydashboard r-leaflet
```
(slightly edited). Note the absence of the r-base=3.5.0 pin.

**Comment by mcg1969** [comment]
@squidnee @nickm-anaconda I'm pretty sure this is hitting us

**Comment by mcg1969** [comment]
The best short-term solution, in my view, is to add the specs with pinned versions to any conda install command. Or to create a pinned spec

**Comment by mcg1969** [comment]
@cjmartian if you want another challenge...

**Comment by mcg1969** [comment]
Addressed by #203. Ultimately I went with pinning over adding the constraints to the conda command itself, because the former does a better job of keeping things undisturbed if the constraints are already satisfied.
